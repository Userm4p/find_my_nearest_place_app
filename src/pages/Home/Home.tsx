import { CityList } from "./components/CityList";
import { Header } from "./components/Header";
import { CitiesContext } from "./context/CitiesContext";
import { useCities } from "./hooks/useCities";

const Home = () => {
  const citiesHookInstance = useCities();

  return (
    <CitiesContext.Provider value={citiesHookInstance}>
      <div className="Home__container">
        <Header />
        <CityList />
      </div>
    </CitiesContext.Provider>
  );
};

export default Home;
