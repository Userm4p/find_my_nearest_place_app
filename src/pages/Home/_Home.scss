.Home__container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100vw;
    gap: 24px;
}