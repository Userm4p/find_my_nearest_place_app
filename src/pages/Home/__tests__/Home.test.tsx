import { render } from '@testing-library/react';
import Home from '../Home';

jest.mock("../hooks/useCities", () => ({
    useCities: () => ({
        citiesToTable: Array.from({ length: 20 }, (_, i) => ({
            name: `City ${i}`,
            country: `Country ${i}`,
            lat: `${i}`,
            lng: `${i}`,
        })),
        pagesNumber: 20,
        page: 1,
        handleNextPage: () => {},
        handlePrevPage: () => {},
        values: { city: "" },
        handleChange: () => {},
        reset: () => {},
        cities: [],
    }),
}));

describe('Tests in Home', () => {
    test('should match snapshot',() => {
            const { container } = render(<Home />);
            expect(container).toMatchSnapshot();
    });
});