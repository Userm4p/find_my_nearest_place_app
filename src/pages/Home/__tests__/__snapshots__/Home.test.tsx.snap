// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Home should match snapshot 1`] = `
<div>
  <div
    class="Home__container"
  >
    <h1
      class="sc-hLQSwg usatO"
    >
      Find My Nearest Place App üåç
    </h1>
    <input
      class="sc-eDLKkx bPRxgU"
      name="city"
      placeholder="Type a city"
    />
    <div
      class="Home__CityList_container"
    >
      <div
        class="Home__CityList_header"
      >
        <div
          class="Home__CityList_header_pages_container"
        >
          <span
            class="sc-dmyCSP edUgyG"
          >
            Page
          </span>
          <span
            class="sc-dmyCSP edUgyG"
          >
            2
          </span>
          <span
            class="sc-dmyCSP edUgyG"
          >
            of
          </span>
          <span
            class="sc-dmyCSP edUgyG"
          >
            20
          </span>
        </div>
        <div
          class="Home__CityList_header_buttons_container"
        >
          <button
            class="sc-guDLey csPTfk"
            data-testid="prev_button"
          >
            Prev
          </button>
          <button
            class="sc-guDLey csPTfk"
            data-testid="next_button"
          >
            Next
          </button>
        </div>
      </div>
      <div
        class="Home__CityList_cities_container"
      >
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 0
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 1
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 2
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 3
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 4
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 5
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 6
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 7
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 8
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 9
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 10
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 11
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 12
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 13
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 14
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 15
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 16
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 17
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 18
        </div>
        <div
          class="sc-beySPh hWTDYJ"
        >
          City 19
        </div>
      </div>
    </div>
  </div>
</div>
`;
